server/src/main/java/org/elasticsearch/index/query/CoordinatorRewriteContext.java

```diff
@@ -53,29 +61,98 @@ public CoordinatorRewriteContext(
             null,
             null
         );
-        this.indexLongFieldRange = indexLongFieldRange;
-        this.timestampFieldType = timestampFieldType;
+        this.dateFieldRangeInfo = dateFieldRangeInfo;
     }
 
-    long getMinTimestamp() {
-        return indexLongFieldRange.getMin();
+    /**
+     * Get min timestamp for either '@timestamp' or 'event.ingested' fields. Any other field
+     * passed in will cause an {@link IllegalArgumentException} to be thrown, as these are the only
+     * two fields supported for coordinator rewrites (based on time range).
+     * @param fieldName Must be DataStream.TIMESTAMP_FIELD_NAME or IndexMetadata.EVENT_INGESTED_FIELD_NAME
+     * @return min timestamp for the field from IndexMetadata in cluster state.
+     */
+    long getMinTimestamp(String fieldName) {
+        if (fieldName.equals(DataStream.TIMESTAMP_FIELD_NAME)) {
```

Comment:
super minor nitpick: fieldName can never be null right? I see that getMaxTimestamp does the equal the other way around, that's not going to throw NPE, while this one is different. Perhaps it does not matter, but I wondered why.

Reply:
Thanks, good catch. I'll make them consistent.